bpf/bpf_sock.c

```diff
@@ -286,6 +286,49 @@ int sock4_xlate(struct bpf_sock_addr *ctx)
 	return SYS_PROCEED;
 }
 
+#ifdef ENABLE_NODEPORT
+__section("bind-sock4")
+int sock4_bind(struct bpf_sock_addr *ctx)
+{
+	struct lb4_service *svc;
+	struct lb4_key key = {
+		.address	= ctx->user_ip4,
+		.dport		= ctx_get_port(ctx),
+	};
+
+	if (!sock_proto_enabled(ctx))
+		return SYS_PROCEED;
+
+	svc = __lb4_lookup_service(&key);
+
+	if (!svc) {
+		/* The k8s watcher already installs service frontends
+		 * for NodePort IPs. If we have not found a service, the
+		 * only remaining overlap might be with the loopback
+		 * address, for which we perform a wildcard lookup.
+		 */
+		__u16 dport = bpf_ntohs(key.dport);
+		if (dport >= NODEPORT_PORT_MIN && dport <= NODEPORT_PORT_MAX &&
+			is_v4_loopback(key.address)) {
```

Comment:
This case doesn't consider any other local address aside the two installed for nodeport. I would simply drop the `is_v4_loopback` to alway check if there is a 0.0.0.0/32 under nodeport port.

Reply:
Thanks for the review! I will adapt your suggestion, will also make the code simpler.

My original thinking was that some `bind()` calls on the nodeport port may not interfere with our NodePort forwarding. But blocking the nodeport port for all IPs will be consistent with kube-proxy, which also binds the port for `0.0.0.0`, so that should be fine.